Bug in dataset generation for tests

I get the following error when running the common tests for the NIX IO:

```python
    def setUp(self):
        '''
            Set up the test fixture.  This is run for every test
            '''
        self.higher = self.ioclass.supported_objects[0]
        self.shortname = self.ioclass.__name__.lower().strip('io')
        # these objects can both be written and read
        self.io_readandwrite = list(set(self.ioclass.readable_objects) &
                                    set(self.ioclass.writeable_objects))
        # these objects can be either written or read
        self.io_readorwrite = list(set(self.ioclass.readable_objects) |
                                   set(self.ioclass.writeable_objects))
        self.create_local_dir_if_not_exists()
        self.download_test_files_if_not_present()
        self.files_generated = []
>       self.generate_files_for_io_able_to_write()
```
```
neo/test/iotest/common_io_test.py:98:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
neo/test/iotest/common_io_test.py:416: in generate_files_for_io_able_to_write
    generate_from_supported_objects(self.ioclass.supported_objects)
neo/test/generate_datasets.py:149: in generate_from_supported_objects
    higher = generate_one_simple_block(supported_objects=objects)
neo/test/generate_datasets.py:41: in generate_one_simple_block
    supported_objects=objects, **kws)
neo/test/generate_datasets.py:113: in generate_one_simple_segment
    labels = labels[(rand(evt_size)*len(labels)).astype('i')]
mtrand.pyx:1347: in mtrand.RandomState.rand (numpy/random/mtrand/mtrand.c:19701)
    ???
mtrand.pyx:856: in mtrand.RandomState.random_sample (numpy/random/mtrand/mtrand.c:15527)
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   TypeError: only integer scalar arrays can be converted to a scalar index

mtrand.pyx:167: TypeError
```

I think this is a numpy warning that got upgraded to an error some time ago (using float scalar arrays as indices, see https://github.com/numpy/numpy/commit/c87c7a612ee7f90281797e447255c402b099c73a).

Simply casting `evt_size` to an integer would resolve this, unless I'm missing something with the `Event` generation in that specific instance. I'm about to open a PR with some fixes for the NIX IO and I could include this fix in that as well if that's OK.

--- a/neo/test/generate_datasets.py
+++ b/neo/test/generate_datasets.py
@@ -107,6 +107,7 @@ def generate_one_simple_segment(seg_name='segment 0',
         for name, labels in iteritems(event_types):
             evt_size = rand()*np.diff(event_size_range)
             evt_size += event_size_range[0]
+            evt_size = int(evt_size)
             labels = np.array(labels, dtype='S')
             labels = labels[(rand(evt_size)*len(labels)).astype('i')]
             evt = Event(times=rand(evt_size)*duration, labels=labels)

